packages/std/src/math/signed_decimal.rs

```diff
@@ -0,0 +1,3060 @@
+        match SignedDecimal::checked_from_ratio(numerator, denominator) {
+            Ok(value) => value,
+            Err(CheckedFromRatioError::DivideByZero) => {
+                panic!("Denominator must not be zero")
+            }
+            Err(CheckedFromRatioError::Overflow) => panic!("Multiplication overflow"),
+        }
+    }
+
+    /// Returns the ratio (numerator / denominator) as a SignedDecimal
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use cosmwasm_std::{SignedDecimal, CheckedFromRatioError};
+    /// assert_eq!(
+    ///     SignedDecimal::checked_from_ratio(1, 3).unwrap().to_string(),
+    ///     "0.333333333333333333"
+    /// );
+    /// assert_eq!(
+    ///     SignedDecimal::checked_from_ratio(1, 0),
+    ///     Err(CheckedFromRatioError::DivideByZero)
+    /// );
+    /// ```
+    pub fn checked_from_ratio(
+        numerator: impl Into<Int128>,
+        denominator: impl Into<Int128>,
+    ) -> Result<Self, CheckedFromRatioError> {
+        let numerator: Int128 = numerator.into();
+        let denominator: Int128 = denominator.into();
+        match numerator.checked_multiply_ratio(Self::DECIMAL_FRACTIONAL, denominator) {
+            Ok(ratio) => {
+                // numerator * DECIMAL_FRACTIONAL / denominator
+                Ok(SignedDecimal(ratio))
+            }
+            Err(CheckedMultiplyRatioError::Overflow) => Err(CheckedFromRatioError::Overflow),
+            Err(CheckedMultiplyRatioError::DivideByZero) => {
+                Err(CheckedFromRatioError::DivideByZero)
+            }
+        }
+    }
+
+    /// Returns `true` if the number is 0
+    #[must_use]
+    pub const fn is_zero(&self) -> bool {
+        self.0.is_zero()
+    }
+
+    /// Returns `true` if the number is negative (< 0)
+    #[must_use]
+    pub const fn is_negative(&self) -> bool {
+        self.0.i128() < 0
+    }
+
+    /// A decimal is an integer of atomic units plus a number that specifies the
+    /// position of the decimal dot. So any decimal can be expressed as two numbers.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// # use cosmwasm_std::{SignedDecimal, Int128};
+    /// # use core::str::FromStr;
+    /// // Value with whole and fractional part
+    /// let a = SignedDecimal::from_str("1.234").unwrap();
+    /// assert_eq!(a.decimal_places(), 18);
+    /// assert_eq!(a.atomics(), Int128::new(1234000000000000000));
+    ///
+    /// // Smallest possible value
+    /// let b = SignedDecimal::from_str("0.000000000000000001").unwrap();
+    /// assert_eq!(b.decimal_places(), 18);
+    /// assert_eq!(b.atomics(), Int128::new(1));
+    /// ```
+    #[must_use]
+    #[inline]
+    pub const fn atomics(&self) -> Int128 {
+        self.0
+    }
+
+    /// The number of decimal places. This is a constant value for now
+    /// but this could potentially change as the type evolves.
+    ///
+    /// See also [`SignedDecimal::atomics()`].
+    #[must_use]
```

Comment:
Maybe additional note to `#[must_use]` when neccessary?

Reply:
I think this one is also clear from function name and return type